### PR TITLE
Add development guidelines & small kitchen dev environment

### PR DESCRIPTION
**Pull request by kbogtob**
### What does this PR do?

It adds some development guidelines in the README.md, it adds a CONTRIBUTING.md file and last but not least, it adds a Gemfile with a small kitchen development environment.

### Motivation

First, on boarding on the project might not seem easy if you are not used with Ansible. Moreover, the project lacks automated tests. So it is hard to know you added regressions or not. Hence the addition of documentation to have more leads and hints on how you can work on the project. I also added integration tests so people adding features and fixes can develop them in TDD, or at least, have them tested out.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Contributing to Datadog Agent
```
Comment (albertvaka) [inline CONTRIBUTING.md]:
```suggestion
# Contributing to the Datadog Ansible role
```

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+source 'https://rubygems.org'
+
+group :development do
+  gem 'rake'
+  gem 'chef', '= 14.10.9'
```
Comment (albertvaka) [inline Gemfile:5]: Wrong Gemfile?

Comment (kbogtob) [reply Gemfile:5]: No, no, chef is used for serverspecs.

File: CONTRIBUTING.md (hunk 1 above)
Comment (kbogtob) [reply CONTRIBUTING.md]: Oops! Thanks

File: Gemfile (hunk 2 above)
Comment (albertvaka) [reply Gemfile:5]: 👍makes sense

**Review by albertvaka — APPROVED** [review]
LGTM! 🚢

**Pull request merged**